### ISSUE TITLE
Rust doesn't support x₁ and x₂ identifiers while C++ does.

### ISSUE THREAD

**Issue by VorfeedCanal** [rust-lang/rfcs#3402]

**Comment by thomcc** [comment]
How likely are things to change from the draft before finalization? Is it worth waiting for the UAX31 update to exit draft status?

**Comment by Manishearth** [comment]
Unlikely, I think.

**Comment by Aloso** [comment]
I don't think an RFC is necessary for this. The [non-ascii-idents RFC](https://rust-lang.github.io/rfcs/2457-non-ascii-idents.html) says that Rust follows the UAX 31 spec. So when the UAX 31 spec is updated, Rust's parser should be updated accordingly. Or am I missing something?

**Comment by Manishearth** [comment]
No, because these are not a part of the default identifier profile specced in UAX 31: they are an [additional profile that the spec lists](https://www.unicode.org/reports/tr31/proposed.html#Mathematical_Compatibility_Notation_Profile) which users of the spec can choose to follow instead. We have to explicitly make that choice.

When Unicode releases a new version we are absolutely allowed to update our UAX 31 data without an RFC, but this changes the algorithm we are choosing to use.

**Comment by scottmcm** [comment]
Looks like <https://www.unicode.org/reports/tr31/#Standard_Profiles> is ready to go?  If someone wants to write an RFC to use it, they're welcome to.

(Note that this issue is likely to get closed in the near future per #3756.)

**Comment by Rawk** [comment]
There is a new RFC that covers this: #3840 ID_Compat_Math characters allowed in identifiers.
